user-service/src/main/java/uk/gov/gchq/palisade/service/user/web/UserController.java

```diff
@@ -18,34 +18,31 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.core.io.ClassPathResource;
-import org.springframework.core.io.Resource;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.service.UserCacheWarmerFactory;
+import uk.gov.gchq.palisade.service.UserConfiguration;
 import uk.gov.gchq.palisade.service.user.request.AddUserRequest;
 import uk.gov.gchq.palisade.service.user.request.GetUserRequest;
 import uk.gov.gchq.palisade.service.user.service.UserService;
 
-import javax.annotation.PostConstruct;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-
 @RestController
 @RequestMapping(path = "/")
 public class UserController {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(UserController.class);
     private UserService service;
+    private UserConfiguration userConfig;
 
-    public UserController(@Qualifier("userService") final UserService service) {
+    public UserController(@Qualifier("userService") final UserService service, @Qualifier("userConfiguration") final UserConfiguration configuration) {
```

Comment:
Why the qualifiers? are there `UserConfiguration`s other than the `@Bean("userConfiguration")`?

Reply:
On start up Spring creates a StdUserConfiguration bean as it is within the config package of the service. This helps the service know which to use in this situation, especially when starting in the example profile.